File: Demo/MNG-Ads-SDK/InterstitialViewController.m
//
//  InterstitialViewController.m
//  MNG-Ads-SDK-Demo
//
//  Created by Ben Salah Med Amine on 4/3/15.
//  Copyright (c) 2015 MNG. All rights reserved.
//

#import "InterstitialViewController.h"
#import <AVFoundation/AVFoundation.h>


@interface InterstitialViewController (){
    MNGAdsSDKFactory *interstitialAdsFactory;
    BOOL autoDisplay;
}

@end

@implementation InterstitialViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setCornersButton:_showIntersButton];
    [self setCornersButton:_displayIntersButton];
//    [self setCornersButton:_ovelayBtn];
//    [self setCornersButton:_interBtn];

}

-(void)setCornersButton:(UIButton *) sender{
     [sender.layer setBorderWidth:1.0];
     [sender.layer setBorderColor:[UIColor lightGrayColor].CGColor];
     [sender.layer setCornerRadius:sender.frame.size.height / 2];
}
- (IBAction)openMenu:(id)sender {
    [[APP_DELEGATE drawerViewController]openMenu];
}

- (IBAction)openCloseConfig:(UIButton *)sender {
    sender.selected = ! sender.selected;
    self.configView.hidden = sender.selected;
}

- (IBAction)createInters:(UIButton *)sender {
    if (interstitialAdsFactory.isBusy) {
        NSLog(@"Ads Factory is busy");
        return;
    }
    [interstitialAdsFactory releaseMemory];
    interstitialAdsFactory = [[MNGAdsSDKFactory alloc]init];
    interstitialAdsFactory.clickDelegate = self;
    interstitialAdsFactory.interstitialDelegate = self;
    interstitialAdsFactory.viewController = [APP_DELEGATE drawerViewController];
    
    /**
     * MNGPrefence is the same for all ad types
     * @important: your application can be rejected by Apple if you use the device's location only for advertising
     **/
    MNGPreference *preferences = [Utils getTestPreferences];
    [self.activityIndicatorView startAnimating];
    self.displayIntersButton.hidden = YES;
    self.showIntersButton.hidden = YES;
    if (sender.tag == 1) {
        autoDisplay = NO;
        interstitialAdsFactory.placementId = MNG_ADS_INTERSTITIAL_PLACEMENT_ID;
        [interstitialAdsFactory loadInterstitialWithPreferences:preferences autoDisplayed:NO];
    }else{
        autoDisplay = YES;
        
        interstitialAdsFactory.placementId = MNG_ADS_INTERSTITIAL_OVERLAY_PLACEMENT_ID;
        [interstitialAdsFactory loadInterstitialWithPreferences:preferences autoDisplayed:YES];
        
    }
    APP_DELEGATE.drawerViewController.statusBarStyle = UIStatusBarStyleLightContent;
    
}

#pragma mark - MNGAdsAdapterInterstitialDelegate

-(void)adsAdapterInterstitialDidLoad:(MNGAdsAdapter *)adsAdapter{
    NSLog(@"adsAdapterInterstitialDidLoad");
    if (adsAdapter == interstitialAdsFactory) {
        if (autoDisplay) {
            self.showIntersButton.hidden = YES;
            self.displayIntersButton.hidden = YES;
        } else {
            self.showIntersButton.hidden = NO;
            self.displayIntersButton.hidden = NO;
        }
        
    }
    
    [self.activityIndicatorView stopAnimating];
    
}

-(void)adsAdapterInterstitialDisappear:(MNGAdsAdapter *)adsAdapter{
    NSLog(@"adsAdapterInterstitialDisappear");
    
    self.displayIntersButton.hidden = YES;
    self.showIntersButton.hidden = YES;
    
}
-(void)adsAdapterInterstitialDidShown:(MNGAdsAdapter *)adsAdapter{
    NSLog(@"adsAdapterInterstitialDidShown");

}
-(void)adsAdapter:(MNGAdsAdapter *)adsAdapter interstitialDidFailWithError:(NSError *)error{
    
    NSLog(@"%@",error);
    [self.activityIndicatorView stopAnimating];
}

-(void)adsAdapterAdWasClicked:(MNGAdsAdapter *)adsAdapter {
    if (adsAdapter == interstitialAdsFactory) {
        NSLog(@"----->Inter1 clicked");
    }
    
}


- (IBAction)showInters:(UIButton*)sender {
    if ([interstitialAdsFactory isInterstitialReady]) {
        if(sender.tag == 0){
            [interstitialAdsFactory showAdFromRootViewController:[APP_DELEGATE drawerViewController] animated:YES];
        }else if (sender.tag == 1){
            [interstitialAdsFactory displayInterstitial];
        }
        
    }
}
-(void)dealloc{
 
    [interstitialAdsFactory releaseMemory];
    interstitialAdsFactory = nil;
  
}
@end

File: DemoSwift/DemoSwift/Interstitial/InterstitialViewController.swift
//
//  InterstitialViewController.swift
//  BlueStackDemoDev
//
//  Created by MedSghaier on 17/8/2022.
//

import UIKit
import BlueStackSDK

class InterstitialViewController: UIViewController , MNGAdsAdapterInterstitialDelegate , MNGClickDelegate {
    
    
    @IBOutlet weak var activityindicatorView: UIActivityIndicatorView!
    
    @IBOutlet weak var configView: UIView!
    
    @IBOutlet weak var showIntersButton: UIButton!
    @IBOutlet weak var displayIntersButton: UIButton!
    
    
    @IBOutlet weak var ovelayBtn: UIButton!
    @IBOutlet weak var interBtn: UIButton!
    
    var interstitialAdsFactory: MNGAdsSDKFactory!
    var autoDisplay : Bool = false
    
    var childVC : ChildInterViewController?
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        interstitialAdsFactory = MNGAdsSDKFactory()
        
        self.showIntersButton.layer.borderWidth = 1
        self.showIntersButton.layer.borderColor = UIColor.lightGray.cgColor
        self.showIntersButton.layer.cornerRadius = self.showIntersButton.frame.size.height / 2
        
        self.displayIntersButton.layer.borderWidth = 1
        self.displayIntersButton.layer.borderColor = UIColor.lightGray.cgColor
        self.displayIntersButton.layer.cornerRadius = self.displayIntersButton.frame.size.height / 2
        
      //  self.addVC()
        
    }

    // MARK: - add View Controller Child
    
    func addVC()  {
        childVC = ChildInterViewController()
        childVC?.view.frame = CGRect(x: 50, y: 270, width: UIScreen.main.bounds.width - 100, height: 290)
        childVC?.view.autoresizingMask = [.flexibleHeight , .flexibleWidth]
        self.view.addSubview(childVC!.view)
        self.addChild(childVC!)
        
    }
  
    // MARK: - OpenMenu button

     
    @IBAction func openMenu(_ sender: Any) {
        APP_DELEGATE.drawerViewController?.openMenu()
    }
    
    
    @IBAction func openCloseConfig(_ sender: UIButton) {
        sender.isSelected = !sender.isSelected
        self.configView.isHidden = sender.isSelected
    }
    
    
    // MARK: - Inters Button Actions
   
    @IBAction func showInters(_ sender: UIButton) {
        if interstitialAdsFactory.isInterstitialReady() {
            if sender.tag == 0 {
                interstitialAdsFactory.showAd(fromRootViewController: APP_DELEGATE.drawerViewController, animated: true)
            } else if sender.tag == 1 {
                interstitialAdsFactory.displayInterstitial()
            }
        }
    }
    
   
    // MARK: - Inters Button Actions
    @IBAction func createInters(_ sender: UIButton) {
        
        if interstitialAdsFactory.isBusy {
             print("Ads Factory is busy")
             interstitialAdsFactory.releaseMemory()
            return
        }
        
        interstitialAdsFactory.interstitialDelegate = self
        interstitialAdsFactory.clickDelegate = self
        interstitialAdsFactory.viewController = self

        let preferencess = MNGPreference.init()
        
        self.activityindicatorView.startAnimating()
        self.displayIntersButton.isHidden  = true
        self.showIntersButton.isHidden = true
        
        if sender.tag == 1 {
            autoDisplay = false
            interstitialAdsFactory.placementId = "/\(MNG_ADS_APP_ID!)\(DemoSwiftConstants.PLACEMENTS.MNG_ADS_INTERSTITIAL_PLACEMENT_ID)"
            interstitialAdsFactory.loadInterstitial(withPreferences: preferencess, autoDisplayed: false)
        } else {
            autoDisplay = true
            interstitialAdsFactory.placementId = "/\(MNG_ADS_APP_ID!)\( DemoSwiftConstants.PLACEMENTS.MNG_ADS_INTERSTITIAL_OVERLAY_PLACEMENT_ID)"
            interstitialAdsFactory.loadInterstitial(withPreferences: preferencess, autoDisplayed: true)
        }
        
        APP_DELEGATE.drawerViewController?.statusBarStyle = .lightContent
    }
    
    
    // MARK: - MNGAdsAdapterInterstitialDelegate
    
    func adsAdapterInterstitialDidLoad(_ adsAdapter: MNGAdsAdapter!) {
        NSLog("adsAdapterInterstitialDidLoad")
        if autoDisplay {
            self.showIntersButton.isHidden = true
            self.displayIntersButton.isHidden = true
        } else {
            self.showIntersButton.isHidden = false
            self.displayIntersButton.isHidden = false
        }
       
        self.activityindicatorView.stopAnimating()
    }
    
    func adsAdapterInterstitialDisappear(_ adsAdapter: MNGAdsAdapter!) {
        NSLog("adsAdapterInterstitialDisappear")
        self.showIntersButton.isHidden = true
        self.displayIntersButton.isHidden = true
    }
    
    func adsAdapterInterstitialDidShown(_ adsAdapter: MNGAdsAdapter!) {
        NSLog("adsAdapterInterstitialDidShown");
    }
    
    func adsAdapter(_ adsAdapter: MNGAdsAdapter!, interstitialDidFailWithError error: Error!) {
        NSLog("\(String(describing: error))")
        self.activityindicatorView.stopAnimating()
    }
    
    func adsAdapterAdWasClicked(_ adsAdapter: MNGAdsAdapter!) {
        NSLog("----->Inter1 clicked")
    }
    
    
    
    
    
    
}
